Sources/AsyncHTTPClient/ConnectionPool/State Machine/HTTPConnectionPool+HTTP1StateMachine.swift

```diff
@@ -323,9 +323,10 @@ extension HTTPConnectionPool {
 
         mutating func cancelRequest(_ requestID: Request.ID) -> Action {
             // 1. check requests in queue
-            if self.requests.remove(requestID) != nil {
+            if let request = self.requests.remove(requestID) {
+                let error = self.lastConnectFailure ?? HTTPClientError.cancelled
```

Comment:
In theory we could check here if we are at max connections... In those cases we could issue a more explicit: `deadlineExceededBecauseToQueueDepthToLong` error.

Reply:
Not sure how useful this kind of error message would be because `deadlineExceededBecauseToQueueDepthToLong` may only be a symptom as other requests in front could not be scheduled because there was a connection failure.